core/src/test/java/com/arcrobotics/ftclib/util/TimingTest.java

```diff
@@ -0,0 +1,87 @@
+package com.arcrobotics.ftclib.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static com.arcrobotics.ftclib.util.Timing.*;
+
+public class TimingTest {
+    private Timer timer;
+    private final int seconds = 3;
+
+    private void assertElapsedWithin(long minMillis, long margin, Runnable runnable) {
+        Thread curr = Thread.currentThread();
+        Thread interrupt = new Thread(() -> {
```

Comment:
I'd suggest adding a javadoc or comment right before this method explaining what it does. Simply for lookback purposes.